src/workflows/airqo_etl_utils/datautils.py

```
@@ -217,9 +217,10 @@ def extract_devices_data(
             device_category = DeviceCategory.LOWCOST
 
         devices, keys = DataUtils.get_devices(device_category, device_network)
+
         # Temporary fix for mobile devices - # TODO: Fix after requirements review
-        if is_mobile_category and "assigned_grid" in devices.columns:
-            devices = devices[devices["assigned_grid"].apply(has_valid_dict)]
+        if is_mobile_category:
+            devices = devices[devices["mobility"] == True]
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `devices["mobility"]:` for truth checks

Replace with `devices["mobility"]`

(E712)